test/contracts/polygonZkEVM.test.js

```diff
@@ -70,8 +70,8 @@ describe('Polygon ZK-EVM', () => {
         if ((await upgrades.admin.getInstance()).address !== '0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0') {
             firstDeployment = false;
         }
-        const nonceProxyBridge = Number((await ethers.provider.getTransactionCount(deployer.address))) + (firstDeployment ? 3 : 2);
-        const nonceProxyZkevm = nonceProxyBridge + 2; // Always have to redeploy impl since the polygonZkEVMGlobalExitRoot address changes
+        const nonceProxyBridge = Number((await ethers.provider.getTransactionCount(deployer.address))) + (firstDeployment ? 2 : 2);
```

Comment:
What's going on here?

Reply:
We need to precalculate the nonces for checking the address. If its a first-time deployment, the behavior is different than later, as the implementation in (deployProxy(impl)) needs to additionally deploy the implementation contract.
Hence, we need to change the nonces. 

( I tried to make it easier readable and so that it also works if one of the tests is marked with "only". in the commit 716e742). hope this is sufficiently good for merging - in the end this is not really important, just a precalculation)